File: frontend/src/types/timeline.ts
export interface TimelineEvent {
  id: string;
  case_id: string;
  timestamp: string;
  title: string;
  description: string;
  event_type: string;
  source_ids: string[];
  confidence: number;
  evidence_file_name?: string | null;
  attributes: Record<string, unknown>;
}

export interface TimelineResponse {
  case_id: string;
  total_events: number;
  events: TimelineEvent[];
}

File: backend/app/schemas/timeline.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, ConfigDict


class TimelineEvent(BaseModel):
    id: str
    case_id: str
    timestamp: datetime
    title: str
    description: str
    event_type: str  # "EXTRACTED_EXIF", "CHAT_MESSAGE", "CALL_LOG", "OBSERVATION", "INGESTION"
    source_ids: List[str]
    confidence: float = 1.0
    evidence_file_name: Optional[str] = None
    attributes: Dict[str, Any] = {}

    model_config = ConfigDict(from_attributes=True)


class TimelineResponse(BaseModel):
    case_id: str
    total_events: int
    events: List[TimelineEvent]
